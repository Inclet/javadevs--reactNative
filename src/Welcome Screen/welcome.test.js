import React from "react";
import { shallow } from "enzyme";
import Welcome from './index';

describe('Welcome Screen', () => {
  describe('Rendering', () => {
    it('Should match Snapshot', () => {
      const component = shallow(<Welcome />);
      expect(component).toMatchSnapshot();
    });

    it("Should Navigate to About us screen when clicking on an arrow", () => {
      const props = {
        navigation: {
          navigate: jest.fn(),
        },
      };
      const welcome = shallow(<Welcome {...props} />);
      welcome.find('[data-testID="nextNavigator"]').simulate("press");
      expect(props.navigation.navigate).toHaveBeenCalledWith("AboutUs");
    });

    it("Should navigate to aboutUs screen when clicking on bottom paginator", () => {
      const props = {
        navigation: {
          navigate: jest.fn(),
        },
      };
      const welcome = shallow(<Welcome {...props} />);
      welcome.find('[data-testID="bottomNavigator"]').simulate("press");
      expect(props.navigation.navigate).toHaveBeenCalledWith("AboutUs");
    });
  });
});
