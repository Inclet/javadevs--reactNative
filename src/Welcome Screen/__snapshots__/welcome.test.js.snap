// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome Screen Rendering Should match Snapshot 1`] = `ShallowWrapper {}`;
